src/ui/VirtualRenderers/VirtualFileRenderer.tsx

```diff
@@ -225,9 +242,28 @@ function VirtualFileRendererComponent({
   const widthDivRef = useRef<HTMLDivElement>(null)
   const codeDisplayOverlayRef = useRef<HTMLDivElement>(null)
   const textAreaRef = useRef<HTMLTextAreaElement>(null)
+  const scrollBarRef = useRef<HTMLDivElement>(null)
   const virtualCodeRendererRef = useRef<HTMLDivElement>(null)
+  const { wrapperWidth, setWrapperRefState } = useSyncWrapperWidth()
+
+  // disable pointer events will scrolling
   useDisablePointerEvents(virtualCodeRendererRef)
-  useLeftScrollSync({ textAreaRef, overlayRef: codeDisplayOverlayRef })
+
+  // sync the width of the wrapper with the width of the text area
+  useSyncTotalWidth({ textAreaRef, widthDivRef })
+
+  // check if the code display overlay is overflowing, so we can conditionally render the scroll bar
+  const isOverflowing = useIsOverflowing(codeDisplayOverlayRef)
+
+  // sync the scroll position of the text area with the code display overlay and scroll bar
+  useSyncScrollLeft({
+    scrollingRef: textAreaRef,
+    refsToSync: [codeDisplayOverlayRef, scrollBarRef],
+  })
+  useSyncScrollLeft({
+    scrollingRef: scrollBarRef,
+    refsToSync: [codeDisplayOverlayRef, textAreaRef],
```

Comment:
This implies it's not possible to scroll the codeDisplayOverlayRef manually, right?

Reply:
Yes, the `textarea` overlays it, so that's what the user interacts with.

Reply:
Cool ok! just wanted to sanity check my understanding tyty

Reply:
They never actually "touch" the virtualized/tokenized code.